.github/workflows/tests.yml

```diff
@@ -45,7 +45,20 @@ jobs:
       # https://github.com/zf1s/zf1/pull/6#issuecomment-495397170
       - name: Validate composer.json for all packages
         run: |
-          for json in packages/*/composer.json; do COMPOSER=$json composer validate || touch failed; done
+          for json in composer.json packages/*/composer.json; do COMPOSER=$json composer validate || touch failed; done
           test ! -f failed
 
+      - name: Cache Composer packages
+        id: composer-cache
+        uses: actions/cache@v2
+        with:
+          path: vendor
+          key: ${{ runner.os }}-php-${{ hashFiles('**/composer.lock') }}
```

Comment:
as the composer.lock is not tracked in git, not sure if should use composer.json here instead?

as the docs don't mention that:
- https://github.com/actions/cache/blob/9f3a4d3e65eb0e37d28bc055c0d1a91eac6d76fe/examples.md#php---composer

but I've seen some examples saying to use package.json.

probably running this action several times gives the answer....

Reply:
here's some discussion on the not committed lock file:
- https://github.com/bamarni/composer-bin-plugin/issues/50#issuecomment-604512815

Reply:
I think pointing to `composer.json` would make sense if versions specified in composer.json were fixed. (no version ranges)
E.g. in case new version of `zf1s/phpunit` is released, `actions/cache` may still take previous version out of its cache based on the hash of `composer.json`.

Does it make sense to generate and commit lockfile here? Then, when a new version of `zf1s/phpunit` is released, another commit will have to be pushed to `zf1s/zf1` to update the version in the lockfile.
Is there any other option? 🤔

Reply:
My original point was that if composer.json has changes, and it's hash not used for actions/cache, the vendor would not be updated.

If you use composer.json without locking, it's expected to have a cache with a variable contents.

I'm not sure about committing a lock file here. I'll give it a thought.

Reply:
## Lock file for composer.json

There's a common misconception whether `composer.lock` needs to be committed to the repository or not.

It can be summarized with two opposite scenarios:

- projects: [keep lock] in projects where it really matters (production is deployed from it):
- libraries: [do not keep lock] for libraries

[keep lock]: https://getcomposer.org/doc/01-basic-usage.md#commit-your-composer-lock-file-to-version-control
[do not keep lock]: https://getcomposer.org/doc/02-libraries.md#lock-file

As the current composer.json contains only one dependency, let's make it strict (no version ranges) and do not commit lock. as the lock would be just tracking phpunit dependencies, that's silly in my opinion.

And we can rethink this in the future at any time.

As for any other option: don't use `actions/cache` and such problem won't exist :)

Reply:
I think it's a good idea to lock dependencies to exact versions in composer.json in this case.

Reply:
Btw how much time is saved in a run when composer cache is hit?

Reply:
please look at the changes, it's already implemented

Reply:
I saw them, only confirmed that's a good idea :) Thanks.

Reply:
> 
> 
> Btw how much time is saved in a run when composer cache is hit?

To answer my question: I checked https://github.com/glensc/php-zf1s/actions/runs/466000519 and I could see composer install takes 5-10 seconds, when cache is not hit, so in our (simple) case with only 2 dependencies maybe it's not worth to add additional complexity here 🤔

Also something is wrong with `hashFiles`: the hash is missing from the cache key
![image](https://user-images.githubusercontent.com/14310995/103999832-b3038600-519d-11eb-8305-2bf8f93365e4.png)

Reply:
Removed all cache related changes

Reply:
Altho I personally would stick using the cache, then CI would not have to make requests outside its infrastructure. But then again I would like this merged, so all cache logic is removed now.